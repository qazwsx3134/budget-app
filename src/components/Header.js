import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from "react-redux";
import { logout } from "../actions/auth";
import Button from '@material-ui/core/Button';
import { apiUserLogout } from "../api/api";
import { useDispatch } from 'react-redux'


export const Header = (props) => {
  const dispatch = useDispatch()

  async function onlogout() {
    // You can await here
    const logoutRes = await apiUserLogout()
    

    dispatch(logout())
    
    console.log('log out');
    
    // ...
  }
  
  return(
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/" >
          <h2>Wayfarer</h2>
        </Link>


        {props.auth._id 
        ?
        <div>
        <Button size="small" color="secondary">Welcome Back , {props.auth.firstName} </Button>
        <Link className="header__title" to="/" >
        <Button color="primary" variant="outlined" onClick={onlogout}>
            LogOut
          </Button>
        </Link>
        
        </div>
        :
        
        <Link className="header__title" to="/login" >
        <Button color="primary" variant="outlined">
            Login
          </Button>
        </Link>}


        {props.auth.isAdmin
        ?
        <Link  className="header__title" to="/create" >
        <Button color="primary" variant="outlined">
            Create
          </Button>
        </Link>
        : null}
        

      </div> 
    </div>
  </header>
)};

const mapStateToProps = (state)=>({ //把action 的dispatch 和header map再一起
  auth: state.auth
});

export default connect(mapStateToProps)(Header);
