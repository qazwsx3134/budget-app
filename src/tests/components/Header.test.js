import { shallow } from "enzyme";
import Header from "../../components/Header";
import React from "react";

test('should render header right', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot(); 

    // expect(wrapper.find('h1').text()).toBe('Expensify')
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />)
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
}) 