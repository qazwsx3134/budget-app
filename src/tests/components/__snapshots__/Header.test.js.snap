// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header right 1`] = `
<header>
  <h1>
    Expensifydddd
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Create Expense
  </NavLink>
</header>
`;
